Use the standard menu option for all map page refreshes

eskyuu opened the pull request:
This removes the new page refresh countdown from the world map and enables the menu based refresh countdown in the top-right menu for all maps.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

eskyuu committed:
Use the standard menu option for all map page refreshes

murrant committed:
Consolidate code into refresh-timer component

murrant committed:
No need to rename variables

eskyuu committed:
Add missing refreshMap() on page load for the device dependency page
Fix the title on the netmap page
Make initial page load code the same on all pages

murrant commented:
Don't need to async+await if we are only calling refreshMap()... The only reason I called it that way was to wait for it to finish before starting the refresh timer on the netmap...

eskyuu commented:
> Don't need to async+await if we are only calling refreshMap()... The only reason I called it that way was to wait for it to finish before starting the refresh timer on the netmap...

The main reason for the change was to add in the missing refreshMap() on page load for the device dependency page.

I added the async+await to make all pages the same.  Are you saying that it is only needed on one of the pages?

laf reviewed (dismissed):
Tested fine on demo.

murrant commented:
Correct, only needed on pages where we call Countdown.Start() after the first data load.

Also, it looks like you removed those. Does this break netmap when there are more than 500 devices?

murrant commented:
async: false is pretty bad on the ajax calls as it blocks the ui thread, basically freezing the interface while the call runs.
That is why I did the async/await call as I did so it synchronizes within the function but does not block the ui thread.

Additionally, this does not refresh when devices > 500.  Let me see if I can get those fixed up.

eskyuu committed:
Remove the async/await on the document ready functions

eskyuu commented:
Thanks - that makes sense regarding why netmap uses async/await.  I needed to add async: false to the netmap otherwise we don't have a list of nodes and edges when the map is created, and the initial placement doesn't work.  It was especially needed on the netmap so we know which nodes are not linked to anything so they can be hidden.

I added the 500 node limit on auto refresh to stop constant high CPU usage, which is also why I added the warning.  Should this be a config option instead of a hard-coded limit?

murrant committed:
netmap remove ajax async false
blocks the ui thread.  Use await to make sure the two fetches are synchronized with each other (and the timer resume)
Remove extra unneeded code from other maps.
Pause everytime for netmap and device-dependency, it won't matter for shorter loads.

murrant committed:
Don't start timer on resume if refresh = 0

murrant commented:
I changed it to always pause in netmap and dep map. It functions the same with easier logic.

Merging this, I'll fix any further bugs in a separate PR>
